lib/teiserver/lobby/libs/lobby_restrictions.ex

```diff
@@ -0,0 +1,282 @@
+defmodule Teiserver.Lobby.LobbyRestrictions do
+  @moduledoc """
+  Helper methods for lobby policies
+  """
+  alias Teiserver.CacheUser
+  require Logger
+  alias Teiserver.Battle.BalanceLib
+  alias Teiserver.Battle
+
+  @rank_upper_bound 1000
+  @rating_upper_bound 1000
+  @splitter "---------------------------"
+  @spec rank_upper_bound() :: number
+  def rank_upper_bound, do: @rank_upper_bound
+  def rating_upper_bound, do: @rating_upper_bound
+
+  def get_lobby_restrictions_welcome_text(state) do
+    play_level_bounds = get_rating_bounds_text(state)
+    play_rank_bounds = get_rank_bounds_text(state)
+
+    cond do
+      play_level_bounds == nil && play_rank_bounds == nil -> []
+      true -> ["This lobby has the following play restrictions:", play_level_bounds, play_rank_bounds]
+      |> Enum.filter(fn x-> x != nil end)
+    end
+  end
+
+  def get_rating_bounds_text(state) do
+    get_rating_bounds_for_title(state)
+  end
+
+  def get_rank_bounds_text(state) do
+    get_rank_bounds_for_title(state)
+  end
+
+  def get_rank_bounds_for_title(consul_state) when consul_state == nil do
+    nil
+  end
+
+  def get_rank_bounds_for_title(consul_state) do
+    default_state = %{maximum_rank_to_play: @rank_upper_bound, minimum_rank_to_play: 0}
+    consul_state = Map.merge(default_state, consul_state)
+    # Chevlevel stuff here
+    cond do
+      # Default chev levels
+      consul_state.maximum_rank_to_play >= @rank_upper_bound &&
+          consul_state.minimum_rank_to_play <= 0 ->
+        nil
+
+      # Just a max rating
+      consul_state.maximum_rank_to_play < @rank_upper_bound &&
+          consul_state.minimum_rank_to_play <= 0 ->
+        "Max chev: #{consul_state.maximum_rank_to_play + 1}"
+
+      # Just a min rating
+      consul_state.maximum_rank_to_play >= @rank_upper_bound &&
+          consul_state.minimum_rank_to_play > 0 ->
+        "Min chev: #{consul_state.minimum_rank_to_play + 1}"
+
+      # Chev range
+      # It shouldn't go here
+      consul_state.maximum_rank_to_play < @rank_upper_bound ||
+          consul_state.minimum_rank_to_play > 0 ->
+        "Chev between: #{consul_state.minimum_rank_to_play} - #{consul_state.maximum_rank_to_play}"
+
+      true ->
+        nil
+    end
+  end
+
+  def get_rating_bounds_for_title(consul_state) when consul_state == nil do
+    nil
+  end
+
+  def get_rating_bounds_for_title(consul_state) do
+    default_state = %{maximum_rating_to_play: @rating_upper_bound, minimum_rating_to_play: 0}
+    consul_state = Map.merge(default_state, consul_state)
```

Comment:
You shouldn't adjust the state this way, just so you can use the `.` accessor later. I think the consul state should have these properties already, but that may be a bit troublesome to check now. So perhaps leave a TODO.

In the meantime, something more idiomatic is to bind these default to variable and use `Map.get/3`:
```
max_rating_to_play = Map.get(consul_state, :maximum_rating_to_play, @rating_upper_bound)
# and similar for the minimum bound
```